services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -495,12 +523,12 @@ getUserStatus uid tid = do
            . zUser uid . zConn "conn"
            . zType "access"
 
-approveLegalHoldDevice :: HasCallStack => UserId -> TeamId -> TestM ResponseLBS
-approveLegalHoldDevice uid tid = do
+approveLegalHoldDevice :: HasCallStack => UserId -> UserId -> TeamId -> TestM ResponseLBS
+approveLegalHoldDevice zusr uid tid = do
     g <- view tsGalley
     put $ g
-           . paths ["teams", toByteString' tid, "legalhold", "approve"]
-           . zUser uid . zConn "conn"
+           . paths ["teams", toByteString' tid, "legalhold", toByteString' uid, "approve"]
```

Comment:
hum, i deliberately removed the `uid` this from the path because it is redundant.  why did you add it again?  (not a big deal, just wondering.)

Reply:
I wasn't aware you'd removed it intentionally; I'm just following the spec on https://github.com/wearezeta/backend-issues/issues/826 so I just assumed this was a typo. 

Also; I really consider it bad API design to have conflicting routes in which one section may be static OR dynamic; e.g.:

- `teams/:tid/legalhold/:uid` 
- `teams/:tid/legalhold/approve`

Having both of these means that re-ordering the route matchers in the routing file WILL break the system, since `approve` will be matched to `:uid`. For this reason I'd prefer to keep it consistent 😄 

I do a check in the handler that the two uid's are the same, so it doesn't allow any strange behaviour 👍

Reply:
very good point, about the hybrid dynamic-static path segment.